test/jdk/java/foreign/TestLayoutPaths.java

```diff
@@ -134,6 +135,38 @@ public void testByteOffsetHandleBadRange() {
         seq.byteOffsetHandle(sequenceElement(5, 1)); // invalid range (starting position is outside the sequence)
     }
 
+    @Test
+    public void testBadAlignmentOfRoot() throws Throwable {
+        MemoryLayout struct = MemoryLayout.structLayout(
+            JAVA_INT,
+            JAVA_SHORT.withName("x"));
+        assertEquals(struct.byteAlignment(), 4);
+
+        try (Arena arena = Arena.ofConfined()) {
+            MemorySegment seg = arena.allocate(struct.byteSize() + 2, struct.byteAlignment()).asSlice(2);
+            assertEquals(seg.address() % JAVA_SHORT.byteAlignment(), 0); // should be aligned
+            assertNotEquals(seg.address() % struct.byteAlignment(), 0); // should not be aligned
+
+            String expectedMessage = "Target offset incompatible with alignment constraints: " + struct.byteAlignment();
+
+            try {
```

Comment:
Can assertThrows be used to simplify this pattern?

Reply:
Yeah, somewhat. I thought `assertThrows` was only a junit thing, but it looks like TestNG also has an `expectThrows` which does the same and returns the thrown exception so that we can check the message.